### PR TITLE
adds architecture diagram, build steps and misc details in README

### PR DESCRIPTION
**Pull request by JBAhire**
This PR,
- explains setting up Hypertrace UI locally
- Adds UI architecture diagram
- Adds list of technologies used
- Adds build steps and image source in README.

`TODO:` I will work with @skjindal93 to add details around architecture in next revision.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,16 +9,54 @@ Install Node + NPM
...
+
+    git pull --recurse-submodules
+
+Pull all changes for the submodules
+
+    git submodule update --remote
+
+## Building Image locally
+
+Hypertrace UI uses gradlew to compile/install/distribute. Gradle wrapper is already part of the source code. To build Hypertrace UI image, run:
```
Comment (aaron-steinfeld) [inline README.md:54]: Gradle is only used to create an image. Compilation happens via npm.

File: README.md [hunk 2]
```diff
@@ -9,16 +9,54 @@ Install Node + NPM
...
+
+```
+./gradlew clean build dockerBuildImages
```
Comment (aaron-steinfeld) [inline README.md:57]: `./gradlew dockerBuildImages` is sufficient. There's no need to specify intermediate targets and clean should never be required to build, gradle's smart enough to rebuild/reuse as appropriate.

File: README.md [hunk 3]
```diff
@@ -9,16 +9,54 @@ Install Node + NPM
...
+  `npm install`
+
+- Initialize Submodule (https://github.com/hypertrace/hypertrace-core-ui/)
+
+  `git submodule update --init`
 
 ## Development server
 
-Run `npm start` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.
+Run `npm run start` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.
```
Comment (aaron-steinfeld) [inline README.md:22]: why add `run`? Works, but not needed, `npm start` is a built in.

File: README.md (hunk 1 above)
Comment (JBAhire) [reply README.md:54]: so only distribute.

File: README.md [hunk 4]
```diff
@@ -19,6 +25,38 @@ Run `npm start` for a dev server. Navigate to `http://localhost:4200/`. The app
...
+Pull all changes for the submodules
+
+    git submodule update --remote
+
+## Building Image locally
+
+Hypertrace UI uses gradlew to build docker image. Gradle wrapper is already part of the source code. To build Hypertrace UI image, run:
+
+```
+./gradlew build dockerBuildImages
```
Comment (aaron-steinfeld) [inline README.md:57]: `build` isn't needed either (it's a no-op here) - just `dockerBuildImages`

File: README.md [hunk 5]
```diff
@@ -19,6 +25,38 @@ Run `npm start` for a dev server. Navigate to `http://localhost:4200/`. The app
...
+
+1. Angular
+2. Typescript
+3. RxJS
+4. [D3](https://d3js.org/)
+5. [Spectator](https://github.com/ngneat/spectator) (_Unit Testing_)
+
+## UI Architecture
+
+| <img src="https://hypertrace-docs.s3.amazonaws.com/ui-architecture.png" width="400" height="400"/> |
```
Comment (aaron-steinfeld) [inline README.md:38]: We should probably update this diagram, it's outdated from before the HT breakout. but it'll work temporarily.

Comment (JBAhire) [reply README.md:38]: cool. we shall update in next week's revision probably.

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**